src/privatesend-util.cpp

```diff
@@ -27,12 +27,14 @@ CScript CKeyHolder::GetScriptForDestination() const
 
 const CKeyHolder& CKeyHolderStorage::AddKey(CWallet* pwallet)
 {
+    LOCK(cs_storage);
     storage.emplace_back(std::unique_ptr<CKeyHolder>(new CKeyHolder(pwallet)));
     LogPrintf("CKeyHolderStorage::%s -- storage size %lld\n", __func__, storage.size());
     return *storage.back();
```

Comment:
This leaks a reference to an unlocked entry in `storage`. The reference seems to be only used to call `GetScriptForDestination()` on it. Maybe it's better to change this method to return a copy of the pubKey instead?

Reply:
Indeed. `pubKey` is a private member however and since we never actually need anything else but corresponding `CScript` anyway I guess changing this to `return storage.back().GetScriptForDestination();` should do the job.

Reply:
Sounds good :+1:

Reply:
pls see #1840